tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/client/SlowConsumerTest.java

```diff
@@ -244,6 +245,49 @@ public void testSlowConsumerSpared() throws Exception {
    }
 
    @Test
+   public void testSlowConsumerWithBurst() throws Exception {
+      ClientSessionFactory sf = createSessionFactory(locator);
+
+      ClientSession session = addClientSession(sf.createSession(true, true));
+
+      ClientProducer producer = addClientProducer(session.createProducer(QUEUE));
+
+      final int numMessages = 20;
+
+      for (int i = 0; i < numMessages; i++) {
+         producer.send(createTextMessage(session, "m" + i));
+      }
+
+      assertPaging();
+
+      final Queue queue = server.locateQueue(QUEUE);
+
+      queue.getRate();
+
+      logger.info("Creating consumer...");
+
+      ClientConsumer consumer = addClientConsumer(session.createConsumer(QUEUE));
+      session.start();
+
```

Comment:
Little java8 trick:

      Wait.waitFor(() -> queue.getConsumerCount() == 0);


not asking a chnage.. just a tip for future use